src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ClientBaseTests.cs

```diff
@@ -125,31 +125,74 @@ public static void ClientMessageInspector_Verify_Invoke()
+        customBinding.Elements.Add(new TextMessageEncodingBindingElement());
+        customBinding.Elements.Add(new HttpTransportBindingElement());
+
+        MyClientBase client = new MyClientBase(customBinding, new EndpointAddress(Endpoints.HttpSoap12_Address));
+        Assert.Equal(CommunicationState.Created, client.State);
+
+        IWcfServiceGenerated serviceProxy = client.ChannelFactory.CreateChannel();
+        Assert.Equal(CommunicationState.Opened, client.State);
+
         try
         {
-            CustomBinding customBinding = new CustomBinding();
-            customBinding.Elements.Add(new TextMessageEncodingBindingElement());
-            customBinding.Elements.Add(new HttpTransportBindingElement());
+            string result = serviceProxy.Echo("Hello");
+            Assert.Equal("Hello", result);
 
-            MyClientBase client = new MyClientBase(customBinding, new EndpointAddress(BaseAddress.HttpBaseAddress));
-            IWcfServiceGenerated serviceProxy = client.ChannelFactory.CreateChannel();
+            Assert.Equal(CommunicationState.Opened, client.State);
 
-            string result = serviceProxy.Echo("Hello");
-            if (!string.Equals(result, "Hello"))
+            client.Close();
+            Assert.Equal(CommunicationState.Closed, client.State);
+        }
+        finally
+        {
+            if (client != null && client.State != CommunicationState.Closed)
             {
-                errorBuilder.AppendLine(String.Format("Expected response from Service: {0} Actual was: {1}", "Hello", result));
+                client.Abort();
+                Assert.Equal(CommunicationState.Closed, client.State);
             }
         }
-        catch (Exception ex)
+    }
+
+    [Fact]
+    [OuterLoop]
+    public static void ClientBaseOfT_NetTcpTransport_Call()
+    {
+        // This test verifies ClientBase<T> can be used to create a proxy and invoke an operation over Tcp
+        // (request reply over Tcp) 
+
+        // This test verifies ClientBase<T> can be used to create a proxy and invoke an operation over Http
+
+        CustomBinding binding = new CustomBinding(
+                new TextMessageEncodingBindingElement(),
+                new TcpTransportBindingElement());
+
+        MyClientBase client = new MyClientBase(binding, new EndpointAddress(Endpoints.Tcp_CustomBinding_NoSecurity_Text_Address));
+        IWcfServiceGenerated serviceProxy = client.ChannelFactory.CreateChannel();
+
+        try
         {
-            errorBuilder.AppendLine(String.Format("Unexpected exception was caught: {0}", ex.ToString()));
-        }
+            string result = serviceProxy.Echo("Hello");
+            Assert.Equal("Hello", result);
 
-        Assert.True(errorBuilder.Length == 0, string.Format("Test Scenario: ClientBaseOfTCall FAILED with the following errors: {0}", errorBuilder));
+            Assert.Equal(CommunicationState.Opened, client.State);
+
```

Comment:
Nit: testing the communication state seems out of scope for this test.  It would be better to create a separate test to verify communication state as affected by ClientBase<T>.  It wouldn't even need to make a call to the service. The purpose of this current test is really just verifying a synchronous round-trip call via ClientBase<T> succeeds.